    @Test
    public void testGetValidationError() {
        ValidationResult result = new ValidationResult();

        result.add("A", createDummyException(10));
        result.add("A", createDummyException(11));
        result.add("B", createDummyException(20));
        result.add("B", createDummyException(21));
        result.add("C", createDummyException(30));
        result.add(300, createDummyException(40));

        Assert.assertEquals(6, result.getErrorCount());
        Assert.assertEquals(4, result.getValidationErrors().size());

        Assert.assertEquals(1, result.getValidationError("C").getExceptionList().size());
        Assert.assertEquals("C", result.getValidationError("C").getErrorObject());

        Assert.assertEquals(2, result.getValidationError("A").getExceptionList().size());
        Assert.assertEquals(0, result.getValidationError(200).getExceptionList().size());

        Assert.assertEquals(3, result.getValidationErrors(String.class).size());
        Assert.assertEquals(1, result.getValidationErrors(Integer.class).size());
    }

